onnxruntime/contrib_ops/cpu/transformers/greedy_search_impl_gpt.h

```diff
@@ -307,6 +326,23 @@ Status GreedySearchGpt<T, ParametersT>::Execute(const FeedsFetchesManager* init_
     // Increase sequence length after a new token is generated.
     ++current_length;
 
+    // Reorder past state after first run if the GPT subgraph (the one used after the first iteration)
+    // contains DecoderMaskedMultiheadAttention nodes
+    if (iteration_counter == 1 && gpt_subgraph_.has_decoder_masked_multihead_attention_) {
+      int offset = gpt_subgraph_.GetFirstPresentOutputIndex();
+      // We will use the same staging buffer while transposing all the layers' past state
```

Comment:
We will use the same staging buffer for all layers while re-ordering the past state buffers.

This is safe as long as we use the same compute stream to do copying into the staging buffer and transposing from the staging buffer.